pkg/agents/team6/communication.go

```diff
@@ -0,0 +1,254 @@
+	a.identifyNeighbours(msg.SenderID(), msg.SenderFloor())
+}
+
+// Handles AskHP messages the agent receives
+// Returns the agent's HP, unless the agent is a narcissist. In this case, he does not answer.
+func (a *CustomAgent6) HandleAskHP(msg messages.AskHPMessage) {
+	if a.currBehaviour.string() != "Narcissist" {
+		reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), a.HP())
+		a.SendMessage(reply)
+		a.Log("I recieved an askHP message from ", infra.Fields{"senderFloor": msg.SenderFloor(), "myFloor": a.Floor()})
+	}
+	// Try to identify our neighbours
+	a.identifyNeighbours(msg.SenderID(), msg.SenderFloor())
+}
+
+func (a *CustomAgent6) HandleStateHP(msg messages.StateHPMessage) {
+	a.identifyNeighbours(msg.SenderID(), msg.SenderFloor())
+}
+
+// Handles AskFoodTaken messages the agent receives
+// Returns the agent's last food intake, unless the agent is a narcissist. In this case, he does not answer.
+func (a *CustomAgent6) HandleAskFoodTaken(msg messages.AskFoodTakenMessage) {
+	if a.currBehaviour.string() != "Narcissist" {
+		reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), int(a.lastFoodTaken))
+		a.SendMessage(reply)
+		a.Log("I recieved an askFoodTaken message from ", infra.Fields{"senderFloor": msg.SenderFloor(), "myFloor": a.Floor()})
+	}
+	// Try to identify our neighbours
+	a.identifyNeighbours(msg.SenderID(), msg.SenderFloor())
+}
+
+// Handles AskIntendedFoodTaken messages the agent receives
+// Returns the agent's last food intake, which is approximately equal the intended food intake, unless the agent is a narcissist. In this case, he does not answer.
+func (a *CustomAgent6) HandleAskIntendedFoodTaken(msg messages.AskIntendedFoodIntakeMessage) {
+	if a.currBehaviour.string() != "Narcissist" {
+		reply := msg.Reply(a.ID(), a.Floor(), msg.SenderFloor(), int(a.lastFoodTaken))
+		a.SendMessage(reply)
+		a.Log("I recieved an askIntendedFoodTaken message from ", infra.Fields{"senderFloor": msg.SenderFloor(), "myFloor": a.Floor()})
+	}
+	// Try to identify our neighbours
+	a.identifyNeighbours(msg.SenderID(), msg.SenderFloor())
+}
+
+// Handles the messages that needs to be propagated
+// Sends the messages to the target floor, unless the agent is a narcissist
+func (a *CustomAgent6) HandlePropagate(msg messages.ProposeTreatyMessage) {
+	// The Narcissist does not propagate treaties
+	if a.currBehaviour.string() != "Narcissist" {
+		treatyToPropagate := messages.NewProposalMessage(msg.SenderID(), msg.SenderFloor(), msg.TargetFloor(), msg.Treaty())
+		a.SendMessage(treatyToPropagate)
+		a.Log("I propogated a treaty")
+	}
+	// Try to identify our neighbours
+	a.identifyNeighbours(msg.SenderID(), msg.SenderFloor())
+}
+
+// Handles the responses from other agents to our treaty proposals
+func (a *CustomAgent6) HandleTreatyResponse(msg messages.TreatyResponseMessage) {
+	if msg.Response() {
+		// Adds accepted treaty in the active treaties of the agent
+		treaty := a.proposedTreaties[msg.TreatyID()]
+		a.AddTreaty(treaty)
```

Comment:
Should you sign the treaty here too to increase the treaty count?

Reply:
Let me check this one - cheers for spotting <3

Reply:
We interpreted it two ways:

1.) either we sign a treaty then sign it off
2) or we receive a treaty and sign on their behalf

We assume that the former is more logical, assuming that the treaty response we receive is already signed

Reply:
(in which case, we shouldn't have to sign)

Reply:
Fair